File: updater/src/main/java/updater/identifiers/WorldMapAreaData.java
package updater.identifiers;

import updater.utility.AbstractIdentifier;
import updater.utility.ClassWrapper;
import updater.utility.DependsOn;

@DependsOn(WorldMapArea.class)
public class WorldMapAreaData extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.isSuperClassEquals("WorldMapArea");
    }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/disabled/WorldMapAreaData.kt
package org.runestar.client.updater.mapper.identifiers.disabled

import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.annotations.DependsOn
import org.runestar.client.updater.mapper.identifiers.classes.WorldMapArea
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.wrappers.Class2

@DependsOn(WorldMapArea::class)
class WorldMapAreaData : IdentityMapper.Class() {

    override val predicate = predicateOf<Class2> { it.superType == type<WorldMapArea>() }
}